sycl/doc/extensions/proposed/sycl_ext_intel_cslice.asciidoc

```diff
@@ -0,0 +1,274 @@
+processes, and not within a single process.  For example, consider a single
+process that constructs two SYCL queues on cslice sub-device #0.  Kernels
+submitted to these two queues are guaranteed to conflict, both using the same
+set of execution units.  Therefore, if a single process wants to explicitly
+submit kernels to cslice sub-devices and it wants to avoid conflict, it should
+create queues on different sub-devices.  By contrast, consider an example where
+two separate processes create a SYCL queue on cslice sub-device #0.  In this
+case, the device driver virtualizes access to this cslice, and kernels
+submitted from the first process may run on different execution units than
+kernels submitted from the second process.  In this second case, the device
+driver binds the process's requested cslice to a physical cslice according to
+the overall system load.
+
+Note that this extension can be supported by any implementation.  If an
+implementation supports a backend or device without the concept of cslice
+partitions it can still conform to this extension by declaring the new
+enumerator and member functions specified below.  If the info descriptor query
+`info::device::partition_properties` does not report
+`ext_intel_partition_by_cslice`, then the backend or device does not support
+the creation of cslice partitions.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_CSLICE` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== New partition property
+
+This extension adds a new enumerator named `ext_intel_partition_by_cslice` to
+`info::partition_property`:
+
+```
+namespace sycl::info {
+
+enum class partition_property : /* unspecified */ {
+  // ...
+  ext_intel_partition_by_cslice
+};
+
+} // namespace sycl::info
+```
+
+The behavior of the `info::device::partition_properties` info descriptor query
+is also extended to include `ext_intel_partition_by_cslice` in the vector of
+returned values if the device can be partitioned into at least two sub-devices
+along that partition property.
+
+=== New function template specialization to create sub-devices
+
+This extension adds a new function template specialization to the `device`
+class:
+
+```
+namespace sycl {
+
+class device {
+  // ...
+
+  // Available only when
+  // Prop == info::partition_property::ext_intel_partition_by_cslice
+  template <info::partition_property Prop>
+  std::vector<device> create_sub_devices() const;
```

Comment:
Shouldn't it be `ext_intel_create_sub_devices` since this is an extension?

Reply:
@romanovvlad: Does my answer address your concern?